File: ProyectosCSharp/SoftProg/SoftProgBusiness/BO/IBaseBO.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SoftProgBusiness.BO
{
    public interface IBaseBO <T>
    {
        int insertar(T objeto);
        int modificar(T objeto);
        int eliminar(int idObjeto);
        T obtenerPorId(int idObjeto);
        BindingList<T> listarTodos();
        void validar(T objeto);
    }
}

File: Arquitectura_REST_SOAP/Backend/SoftProgBusiness/src/main/java/pe/edu/pucp/softprog/bo/IBaseBO.java
package pe.edu.pucp.softprog.bo;

import java.util.ArrayList;

public interface IBaseBO <T>{
    int insertar(T objeto) throws Exception;
    int modificar(T objeto) throws Exception;
    int eliminar(int idObjeto) throws Exception;
    T obtenerPorId(int idObjeto) throws Exception;
    ArrayList<T> listarTodos() throws Exception;
    void validar(T objeto) throws Exception;
}